vendor/github.com/sn-srdjan/cx-tracker-cli/src/cli/cxtracker/tracker_provider.go

```diff
@@ -0,0 +1,91 @@
+package cxtracker
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+
+	"github.com/sn-srdjan/cx-tracker-cli/src/api"
+)
+
+// Provider - provider object
+type Provider struct {
+	ServiceURL string
+	apiClient  *api.Client
+}
+
+// CXApplicationConfig holds config describing CX Application instance
+type CXApplicationConfig struct {
+	//TODO confirm should config contain genesis hash and work based on it (alternative needing to use config file content hash), and
+	// is genesis hash actually the same thing as genesis signature (hash for the first element in chain)
+
+	// Pro is having Genesis hash clearly displayed for user and using content hash just to uniquelly identify the app, and not having to highlight computed hash to user
+	// Con is potentially having duplicate genesis hashes (if other app related config changes - depends what's left to be added to the config)
+	GenesisHash    string `json:"genesisHash"`
+	GenesisAddress string `json:"genesisAddress"`
+	PublicKey      string `json:"publicKey"`
+	SecretKey      string `json:"secretKey"`
+}
+
+// DefaultCxTrackerURL - default cx tracker url
+const DefaultCxTrackerURL = "https://cx-tracker.skycoin.net/api/v1/config"
+
+// SaveToTrackerService - persist config on tracker service
+func (t *Provider) SaveToTrackerService(configFilePath string) error {
+	t.init()
+	if _, err := os.Stat(configFilePath); os.IsNotExist(err) {
+		return fmt.Errorf("config %s doesn't exist", configFilePath)
+	}
+
+	bs, err := ioutil.ReadFile(configFilePath)
+	if err != nil {
+		return fmt.Errorf("error while reading config %s", configFilePath)
+	}
+
+	r := bytes.NewReader(bs)
+
+	if err := t.apiClient.Put(t.ServiceURL, r, nil); err != nil {
+		return fmt.Errorf("error while persisting config %s on service %s due to error: %s", configFilePath, t.ServiceURL, err)
+	}
+
+	return nil
+}
+
+// GetConfigFromTrackerService pulls config from CX Tracker service by it's genesis hash value
+func (t *Provider) GetConfigFromTrackerService(genesisHash, configFilePath string) error {
+	t.init()
+	t.ServiceURL = t.ServiceURL + "/" + genesisHash + "/file"
+	configResp := CXApplicationConfig{}
+	err := t.apiClient.Get(t.ServiceURL, &configResp)
+	if err != nil {
+		return fmt.Errorf("error while retreiving config with genesis hash: %s on service %s due to error: %s", genesisHash, t.ServiceURL, err)
+	}
+
+	data, err := json.MarshalIndent(configResp, "", " ")
+	if err != nil {
+		return fmt.Errorf("error while marshal config with genesis hash: %s due to error: %s", genesisHash, err)
+	}
+
+	f, err := os.Create(configFilePath)
+	if err != nil {
+		return fmt.Errorf("error while creating file for config with genesis hash: %s due to error: %s", genesisHash, err)
```

Comment:
Should discuss a couple of scenarios here:
- if we don't have rights or parts of the path don't exist we should just fail
- if there is already config file on given path right now it's overwritten - not sure if we should do something about that (keep old as backup or something else)

Reply:
* I don't know how we could not have rights to write to the path. Are you trying to write somewhere in root or somewhere obscure? Then it will be a problem, but this shouldn't be the case. Everything should be written to the user directory or somewhere where we know for sure the user has rights to write to. In case the user did something really weird (which happens more than I ever imagined), yeah, sure, throw an error.

* I'm not sure if I have the whole picture in my mind about this, but I see that `options.configPath` represents a **filename**, which means that the user has the capability of specifying multiple configuration files, each for a different CX chain. If the user decides to create another CX chain with the same config filename, we can assume that the user wants to overwrite any existing config files. It'd be extremely annoying to be testing CX chains and find out later on that you have dozens of backup files of config files that you'll never use again.